{
  "origin": "codeshovel",
  "repositoryName": "hadoop",
  "repositoryPath": "/home/shaiful/research/codeshovel/codeshovel-projects/hadoop/.git",
  "startCommitName": "HEAD",
  "sourceFileName": "JobControl.java",
  "functionName": "failAllJobs",
  "functionId": "failAllJobs___t-Throwable",
  "sourceFilePath": "hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/jobcontrol/JobControl.java",
  "functionStartLine": 278,
  "functionEndLine": 295,
  "numCommitsSeen": 11,
  "timeTaken": 1053,
  "changeHistory": [
    "6eba79232f36b36e0196163adc8fe4219a6b6bf9",
    "a9bf33445e950ae62c16ccc09521fb15e1521faf"
  ],
  "changeHistoryShort": {
    "6eba79232f36b36e0196163adc8fe4219a6b6bf9": "Ybodychange",
    "a9bf33445e950ae62c16ccc09521fb15e1521faf": "Yintroduced"
  },
  "changeHistoryDetails": {
    "6eba79232f36b36e0196163adc8fe4219a6b6bf9": {
      "type": "Ybodychange",
      "commitMessage": "HADOOP-14271. Correct spelling of \u0027occurred\u0027 and variants. Contributed by Yeliang Cang\n",
      "commitDate": "03/04/17 8:13 PM",
      "commitName": "6eba79232f36b36e0196163adc8fe4219a6b6bf9",
      "commitAuthor": "Chris Douglas",
      "commitDateOld": "17/02/14 8:14 AM",
      "commitNameOld": "08e6d312e855b5025211d46aeed6722e9cfac203",
      "commitAuthorOld": "Jason Darrell Lowe",
      "daysBetweenCommits": 1141.46,
      "commitsBetweenForRepo": 8559,
      "commitsBetweenForFile": 1,
      "diff": "@@ -1,18 +1,18 @@\n   synchronized private void failAllJobs(Throwable t) {\n-    String message \u003d \"Unexpected System Error Occured: \"+\n+    String message \u003d \"Unexpected System Error Occurred: \"+\n     StringUtils.stringifyException(t);\n     Iterator\u003cControlledJob\u003e it \u003d jobsInProgress.iterator();\n     while(it.hasNext()) {\n       ControlledJob j \u003d it.next();\n       try {\n         j.failJob(message);\n       } catch (IOException e) {\n         LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n       } catch (InterruptedException e) {\n         LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n       } finally {\n         failedJobs.add(j);\n         it.remove();\n       }\n     }\n   }\n\\ No newline at end of file\n",
      "actualSource": "  synchronized private void failAllJobs(Throwable t) {\n    String message \u003d \"Unexpected System Error Occurred: \"+\n    StringUtils.stringifyException(t);\n    Iterator\u003cControlledJob\u003e it \u003d jobsInProgress.iterator();\n    while(it.hasNext()) {\n      ControlledJob j \u003d it.next();\n      try {\n        j.failJob(message);\n      } catch (IOException e) {\n        LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n      } catch (InterruptedException e) {\n        LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n      } finally {\n        failedJobs.add(j);\n        it.remove();\n      }\n    }\n  }",
      "path": "hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/jobcontrol/JobControl.java",
      "extendedDetails": {}
    },
    "a9bf33445e950ae62c16ccc09521fb15e1521faf": {
      "type": "Yintroduced",
      "commitMessage": "MAPREDUCE-2756. Better error handling in JobControl for failed jobs. Contributed by Robert Evans. \n\n\ngit-svn-id: https://svn.apache.org/repos/asf/hadoop/common/trunk@1164255 13f79535-47bb-0310-9956-ffa450edef68\n",
      "commitDate": "01/09/11 1:24 PM",
      "commitName": "a9bf33445e950ae62c16ccc09521fb15e1521faf",
      "commitAuthor": "Arun Murthy",
      "diff": "@@ -0,0 +1,18 @@\n+  synchronized private void failAllJobs(Throwable t) {\n+    String message \u003d \"Unexpected System Error Occured: \"+\n+    StringUtils.stringifyException(t);\n+    Iterator\u003cControlledJob\u003e it \u003d jobsInProgress.iterator();\n+    while(it.hasNext()) {\n+      ControlledJob j \u003d it.next();\n+      try {\n+        j.failJob(message);\n+      } catch (IOException e) {\n+        LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n+      } catch (InterruptedException e) {\n+        LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n+      } finally {\n+        failedJobs.add(j);\n+        it.remove();\n+      }\n+    }\n+  }\n\\ No newline at end of file\n",
      "actualSource": "  synchronized private void failAllJobs(Throwable t) {\n    String message \u003d \"Unexpected System Error Occured: \"+\n    StringUtils.stringifyException(t);\n    Iterator\u003cControlledJob\u003e it \u003d jobsInProgress.iterator();\n    while(it.hasNext()) {\n      ControlledJob j \u003d it.next();\n      try {\n        j.failJob(message);\n      } catch (IOException e) {\n        LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n      } catch (InterruptedException e) {\n        LOG.error(\"Error while tyring to clean up \"+j.getJobName(), e);\n      } finally {\n        failedJobs.add(j);\n        it.remove();\n      }\n    }\n  }",
      "path": "hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/jobcontrol/JobControl.java"
    }
  }
}